frontend/src/App.js

```diff
@@ -1,33 +1,31 @@
-import React from "react";
-import { BrowserRouter as Router, Switch, Route, Link } from "react-router-dom";
+import React, { useState, useEffect } from "react";
+import { BrowserRouter as Router, Route } from "react-router-dom";
 import "./App.css";
 
-import AddPost from "./components/AddPost";
-import Posts from "./components/Posts";
+import AuthHandler from "./components/AuthHandler";
+import AuthenticatedApp from "./authenticated";
+import { isAuthenticated } from "./utils";
+import NonAuthenticatedApp from "./nonAuthenticated";
 
 function App() {
+  const [isLoggedIn, setIsLoggedIn] = useState(false);
+
+  useEffect(() => {
+    setIsLoggedIn(isAuthenticated());
+  }, []);
+
+  const renderApp = isLoggedIn ? (
```

Comment:
why use hook `isLoggedIn` as the same output you have with 'isAuthenticated' and that is what you're checking here
```suggestion
  const renderApp = IsAuthenticated ? (
```

Reply:
I use setIsLoggedIn to set isLoggedIn in components further down in the tree
if I use only isAuthenticated and delete token in the component down in the tree isAuthenticated will not be called and app will not re-render 

if there is another solution I will implement it

Reply:
👍